app/models/hardware.rb

```diff
@@ -63,8 +63,14 @@ def self.add_elements(parent, xmlNode)
                    .select(:id, :device_type, :location, :address)
                    .collect { |rec| [rec.id, [rec.device_type, rec.location, rec.address]] }
 
-    deletes[:disk] = parent.hardware.disks.select(:id, :device_type, :location)
+    if parent.vendor == "redhat"
+      deletes[:disk] = parent.hardware.disks.select(:id, :device_type, :location)
+                     .collect { |rec| [rec.id, [rec.device_type, "0:#{rec.location}"]] }
```

Comment:
I don't think this is the right way to fix this. Instead of adding `0:` to the location we get from the database, we should just not add the `0:` to the location we add to the XML generated by SSA. Here, you're adding a "redhat" specific check to common code. The XML generation code is already "redhat" specific, and it's doing something it need not do, so it would be much cleaner fixing it on that end.